math/smatrix/inc/Math/CholeskyDecomp.h

```diff
@@ -329,10 +369,10 @@ template<class F> class CholeskyDecompGenDim
     * operator()(int i, int j) for access to elements)
     */
    template<class M> CholeskyDecompGenDim(unsigned N, const M& m) :
-      fN(N), fL(new F[(fN * (fN + 1)) / 2]), fOk(false)
+      fN(N), fL(std::make_unique<F[]>((fN * (fN + 1)) / 2)), fOk(false)
    {
       using CholeskyDecompHelpers::_decomposerGenDim;
-      fOk = _decomposerGenDim<F, M>()(fL, m, fN);
+      fOk = _decomposerGenDim<F, M>()(fL.get(), m, fN);
```

Comment:
```suggestion
      fOk = _decomposerGenDim<F, M>()(fL.data(), m, fN);
```
With the `std::vector`, you can get the underlying C-style array with `data()`.

Reply:
I know. Let's wait what you decide in the comment with my unique_ptr justificaion above.